src/common/QuirksManager.cpp

```diff
@@ -184,6 +184,17 @@ bool QuirksManager::isConversionNeeded(
         LOGI("QuirksManager::%s() forcing internal format to I16 for low latency", __func__);
     }
 
+    // Add quirk for float output on API <21
+    if (isFloat
+            && !isInput
```

Comment:
Is Float supported for input?!

Reply:
I imagine so, however, I'm unable to test that because I don't have a pre-L device and the emulator doesn't support input streams. I figured it was best to only add code which I could test. We (and by that I mean, you :) could always a quirk for input and then test it on that old phone you have.

Reply:
OK, I will add that and test it on my device.